backend/app/services/execution/polling_dispatcher.py

```
@@ -0,0 +1,352 @@
+
+POLL_INTERVAL_SECONDS = 5
+MAX_POLL_COUNT = 720  # 1 hour at 5s intervals
+DEFAULT_GEMINI_BASE_URL = "https://generativelanguage.googleapis.com"
+DEFAULT_DEEP_RESEARCH_AGENT = "deep-research-pro-preview-12-2025"
+THOUGHT_STREAM_TIMEOUT = 30.0  # seconds, timeout per read for thinking progress
+REPORT_STREAM_TIMEOUT = 300.0  # seconds, timeout per read for final report
+
+
+async def dispatch_polling(
+    request: ExecutionRequest,
+    target: ExecutionTarget,
```

Ruff (0.15.2)

[warning] Unused function argument: `target`

(ARG001)